docs/user/upgrading_renku.rst

```diff
@@ -0,0 +1,89 @@
+to these new features without doing anything special. E.g. when we added the ability
+for anonymous (i.e. not logged in) users to launch interactive environments, all
+projects were immediately accessible to anonymous users without the project owner
+needing to change any settings.
+
+.. _renku_cli_upgrade:
+
+Using new ``renku`` CLI features
+--------------------------------
+
+.. note::
+
+  This section is for upgrading the version of ``renku`` CLI installed into
+  the Interactive Environments on RenkuLab for your project. See :ref:`upgrading_local`
+  for upgrading your local machine's version of ``renku``.
+
+When we release a new version of the ``renku`` CLI, you do have to make some
+(minimal) changes to the ``Dockerfile`` in your project to ensure that the
+Interactive Environment on RenkuLab will use the image with the correct version.
+
+The version of the ``renku`` CLI is defined in the base image specified in the
+``FROM`` line (usually line 1) of the ``Dockerfile`` in your repo.
+
+1. Replace this line with the latest available image for the "flavor" you're using,
+   which you can find by:
+
+* checking the `renkulab-docker repo README.md <https://github.com/SwissDataScienceCenter/renkulab-docker/blob/master/README.md>`_
+  for the naming conventions of the "flavor" of image you're updating (e.g. if you're using
+  the "minimal python project", you will look in the ``py`` section)
+* visiting the dockerhub page linked in the naming conventions section in the above ``README``
+* choosing the latest available tagged image
+
+Note that the version of the ``renku`` CLI is defined in the image name. For example,
+``renku/renkulab-py:3.7-renku0.10.3-0.6.2`` has CLI version ``0.10.3``. Please choose
+an image with the renku version specified (e.g. ``-renku0.10.3-`` and not ``-renku-``),
+unless you absolutely need the latest development version of ``renku``, since it makes
+debugging and reproducibility much simpler.
+
+2. Push your changes (either from the GitLab editor, or in a running Interactive
+   Environment), and start up a new Interactive Environment to use this new image.
+
+3. When you start to run ``renku`` commands, you might be asked to call ``renku migrate``.
+   This command ensures that the metadata in your project is updated.
+
+Test drive
+^^^^^^^^^^
+
+If you just want to try out the latest version of ``renku`` without building a new
+image yet, you can follow the ``pipx`` installation instructions from
+:ref:`upgrading your local installation <upgrading_local>`. Installing in this way,
+like any other terminal install, will not hold through the next time you launch.
+
+Keep in mind that your project's metadata will be upgraded when you start running
+the new version ``renku`` commands - don't push these changes if you want to revert
+back to an older version.
+
+You should follow the steps in the previous section to swap out the base image
+once you're satisfied with latest ``renku``.
+
+Lazy upgrade
```

Comment:
Why do you call this a "lazy upgrade"?

Reply:
😅 inspired by the term "lazy loading" ("to defer initialization of an object until the point at which it is needed"); since we prefer people to swap out the image if possible, but that they can "lazily" append the upgrade line to the dockerfile otherwise.

Reply:
That's why I was confused:  it isn't lazily upgraded here in the sense of "lazy loading." I would take that to mean that the version gets upgraded the moment you try to do something that requires the new version, but that's not what is happening.

I think "Lightweight upgrade" would be a better term.

Reply:
well i mean "the point at which it's needed" as a layer in the Dockerfile; but i see that there are multiple interpretations. Lazy has a negative connotation to hopefully discourage people from doing this unless they have to. Lightweight has a positive connotation that might lead people to following it even if they don't have to.

Reply:
In light of the conversation/decisions today, some of this will need to be reworked anyway - should we postpone this PR until then? Or merge it as is and update?

Reply:
i'll update it to reflect the changes!

Reply:
but we might be in an uncanny valley because people with existing projects will have to add content to their dockerfiles & gitlab-ci, which won't be fun~

Reply:
so maybe let's merge and update, so that this version of the docs exists at some point in the history of the docs.